### ISSUE TITLE
Error: Failed to instantiate provider "azurerm" to obtain schema: Unrecognized remote plugin message

### ISSUE THREAD

**Issue by Aschenbecher** [terraform-providers/terraform-provider-azurerm#7326]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform 0.12.20 - 0.12.26
azurerm plugin 2.x (2.13 & 2.14)

### Affected Resource(s)

azurerm_linux_virtual_machine
### Terraform Configuration Files

```hcl
variable "server_name" {
        default = "simulation"
}

locals {
        sim_srv_name = "${var.environment}-${var.server_name}-${terraform.workspace}"
        opt_srv_name = "${var.environment}-optimizer-${terraform.workspace}"
}

# create simulation server instance
resource "azurerm_linux_virtual_machine" "simulation" {

        name                            = "${local.sim_srv_name}-${format("%03d", count.index +1 )}"
        count                           = var.num_servers
        size                            = var.instance_type_sim
        location                        = data.azurerm_resource_group.metabuild.location
        resource_group_name             = data.azurerm_resource_group.metabuild.name
        network_interface_ids           = ["${azurerm_network_interface.sim[count.index].id}"]
        admin_username        = var.username
        admin_password        = var.password
        custom_data           = base64encode(templatefile("${path.module}/files/userdata.tpl", {}))

#       delete_os_disk_on_termination   = true
#       delete_data_disks_on_termination = true
        source_image_id  = data.azurerm_image.simimage.id
        os_disk {
                name              = "${local.sim_srv_name}-${format("%03d", count.index +1 )}-disk"
                caching           = "ReadWrite"
                storage_account_type = "Standard_LRS"
  }
        disable_password_authentication = false
        tags = {
                # Name = "${var.environment}-${var.server_name}-${terraform.workspace}-${format("%03d", count.index +1 )}"
                prometheus_io_scrape = true
                prometheus_io_port = 9100
                optimizer_id = terraform.workspace
                environment = var.environment

        }
}

resource "azurerm_network_interface" "sim" {
        count                           = var.num_servers
        name                            = "${local.sim_srv_name}-${format("%03d", count.index +1 )}-nic"
        location                        = data.azurerm_resource_group.metabuild.location
        resource_group_name             = data.azurerm_resource_group.metabuild.name
        enable_ip_forwarding            = true
        ip_configuration {
                name                            = "${local.sim_srv_name}-${format("%03d", count.index +1 )}-ipconfig"
                subnet_id                       = data.terraform_remote_state.network.outputs.private_sub
                private_ip_address_allocation   = "Dynamic"
  }
}

# Create optimizer instance
resource "azurerm_linux_virtual_machine" "optimizer" {
        name                  = local.opt_srv_name
        location              = data.azurerm_resource_group.metabuild.location
        resource_group_name   = data.azurerm_resource_group.metabuild.name
        network_interface_ids = ["${azurerm_network_interface.optimizer.id}"]
        size               = var.instance_type_optimizer
        admin_username        = var.username
        admin_password        = var.password
        custom_data           = base64encode(templatefile("${path.module}/files/userdata.tpl", {}))
        source_image_id  = data.azurerm_image.simimage.id
        os_disk {
                name              = "${local.opt_srv_name}-disk"
                caching           = "ReadWrite"
                storage_account_type = "Standard_LRS"
        }
        disable_password_authentication = false
  tags = {
        environment = var.environment
        prometheus_io_scrape = true
        prometheus_io_port = 9100
        optimizer_id = terraform.workspace
        environment = var.environment
  }
}

# Create Network interface for optimizer

resource "azurerm_network_interface" "optimizer" {
        name                            = "${local.opt_srv_name}-nic"
        location                        = data.azurerm_resource_group.metabuild.location
        resource_group_name             = data.azurerm_resource_group.metabuild.name
        enable_ip_forwarding            = true
        ip_configuration {
                name                            = "${local.opt_srv_name}-ipconfig"
                subnet_id                       = data.terraform_remote_state.network.outputs.private_sub
                private_ip_address_allocation   = "Dynamic"
  }
}

```

### Debug Output
```hcl
Error: Failed to instantiate provider "azurerm" to obtain schema: Unrecognized remote plugin message: 

This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol.
```

### Panic Output
```hcl
fatal error: fault
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: [signal SIGBUS: bus error code=0x2 addr=0x3adfa71 pc=0x4861d1]
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: goroutine 1 [running, locked to thread]:
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.throw(0x4354107, 0x5)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/panic.go:1116 +0x72 fp=0xc00006d360 sp=0xc00006d330 pc=0x433212
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.sigpanic()
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/signal_unix.go:692 +0x443 fp=0xc00006d390 sp=0xc00006d360 pc=0x449183
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: reflect.name.name(...)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/reflect/type.go:495
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: reflect.(*rtype).String(0x3b830e0, 0x34, 0xc00006d4e8)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/reflect/type.go:753 +0x51 fp=0xc00006d3c8 sp=0xc00006d390 pc=0x4861d1
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: reflect.typesByString(0xc00006d4e8, 0x11, 0x1, 0x3a78968, 0x10)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/reflect/type.go:1725 +0x133 fp=0xc00006d498 sp=0xc00006d3c8 pc=0x48b7a3
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: reflect.(*rtype).ptrTo(0x3be9560, 0x17)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/reflect/type.go:1391 +0xe2 fp=0xc00006d540 sp=0xc00006d498 pc=0x489a82
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: reflect.PtrTo(...)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/reflect/type.go:1376
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.implementsInterface(0x4c02820, 0x3be9560, 0x4c02820, 0x3dce580, 0x0)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:142 +0x11a fp=0xc00006d588 sp=0xc00006d540 pc=0x888b4a
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.validUserType(0x4c02820, 0x3be9560, 0x3be9560, 0x17, 0x1)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:78 +0x303 fp=0xc00006d618 sp=0xc00006d588 pc=0x8887e3
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.userType(0x4c02820, 0x3be9560, 0x3a44b73)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:152 +0x39 fp=0xc00006d660 sp=0xc00006d618 pc=0x888c39
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.newTypeObject(0x3a7089a, 0xa, 0xc000118a80, 0x4c02820, 0x3e9cb60, 0x0, 0x0, 0x0, 0x0)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:530 +0xd49 fp=0xc00006d858 sp=0xc00006d660 pc=0x88b059
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.getType(0x3a7089a, 0xa, 0xc000118a80, 0x4c02820, 0x3e9cb60, 0x487038, 0x3e9cb60, 0x3a70896, 0xe)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:598 +0xbd fp=0xc00006d8e0 sp=0xc00006d858 pc=0x88b99d
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.getBaseType(0x3a7089a, 0xa, 0x4c02820, 0x3eca5a0, 0x0, 0x0, 0x4c02820, 0x3eca5a0)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:585 +0x68 fp=0xc00006d938 sp=0xc00006d8e0 pc=0x88b898
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.newTypeObject(0x3a5d1f8, 0x8, 0xc000118940, 0x4c02820, 0x4089ac0, 0x0, 0x0, 0x0, 0x0)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:536 +0xd9e fp=0xc00006db30 sp=0xc00006d938 pc=0x88b0ae
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.getType(0x3a5d1f8, 0x8, 0xc000118940, 0x4c02820, 0x4089ac0, 0x487038, 0x4089ac0, 0x3a5d1f4, 0xc)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:598 +0xbd fp=0xc00006dbb8 sp=0xc00006db30 pc=0x88b99d
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.getBaseType(0x3a5d1f8, 0x8, 0x4c02820, 0x4089ac0, 0xc00006dca8, 0xc00015acd0, 0x0, 0xc000118940)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:585 +0x68 fp=0xc00006dc10 sp=0xc00006dbb8 pc=0x88b898
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.buildTypeInfo(0xc000118940, 0x4c02820, 0x4089ac0, 0x0, 0x0, 0x0)
2020-06-15T12:51:43.396Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:715 +0x17a fp=0xc00006dd40 sp=0xc00006dc10 pc=0x88c3da
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.getTypeInfo(0xc000118940, 0x4089ac0, 0xc000118940, 0xc000118900)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:702 +0xf2 fp=0xc00006ddb0 sp=0xc00006dd40 pc=0x88c1f2
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.mustGetTypeInfo(0x4c02820, 0x4089ac0, 0x4c02820)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:766 +0x47 fp=0xc00006ddf8 sp=0xc00006ddb0 pc=0x88cd67
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: encoding/gob.init()
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/encoding/gob/type.go:259 +0x1a7e fp=0xc00006de98 sp=0xc00006ddf8 pc=0x89079e
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.doInit(0x775bac0)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:5414 +0x8a fp=0xc00006dec8 sp=0xc00006de98 pc=0x44231a
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.doInit(0x775dc60)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:5409 +0x57 fp=0xc00006def8 sp=0xc00006dec8 pc=0x4422e7
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.doInit(0x7767760)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:5409 +0x57 fp=0xc00006df28 sp=0xc00006def8 pc=0x4422e7
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.doInit(0x7761ea0)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:5409 +0x57 fp=0xc00006df58 sp=0xc00006df28 pc=0x4422e7
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.doInit(0x77514a0)
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:5409 +0x57 fp=0xc00006df88 sp=0xc00006df58 pc=0x4422e7
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.main()
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/proc.go:190 +0x1ce fp=0xc00006dfe0 sp=0xc00006df88 pc=0x43584e
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: runtime.goexit()
2020-06-15T12:51:43.397Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 	/opt/goenv/versions/1.14.2/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00006dfe8 sp=0xc00006dfe0 pc=0x463311
2020/06/15 12:51:43 [TRACE] LoadSchemas: retrieving schema for provider type "terraform"
2020-06-15T12:51:43.398Z [WARN]  plugin: plugin failed to exit gracefully
2020-06-15T12:51:43.402Z [DEBUG] plugin: plugin process exited: path=/storage/metabuild/terraform/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v2.14.0_x5 pid=89 error="exit status 2"
```
### Expected Behavior

We are using Terraform in a docker container. It was working with azurerm plugin Version 1.x.
Since upgrading to Version 2.x its throwing the error above.
With other provider its still working.
On a localmachine with the same Terraform files its working, just happens inside of the docker container

### Actual Behavior

terraform already crashing while Terraform plan or Terraform validate

### Steps to Reproduce

1. Create Docker container with Terraform 
```hcl
FROM ubuntu:18.04
RUN apt-get update && apt-get install -y unzip wget
RUN wget https://releases.hashicorp.com/terraform/0.12.26/terraform_0.12.26_linux_amd64.zip
RUN unzip terraform_0.12.26_linux_amd64.zip && rm terraform_0.12.26_linux_amd64.zip && mv terraform /bin/terraform
```
2 Start Container:

```hcl
docker run -it --name worker-test --network=staging-converter -e NODE_ENV=staging -e DATACENTER=azure --restart=always -v /storage/xxx/:/storage/xxx -v /home/xxx/:/home/xxx/ -v /etc/group:/etc/group -u xxx  registry.xxx.xx/worker:test bash
```
3. Move to the mounted folder with the Terraform files
4. `terraform init`
5. `terraform apply`

### Important Factoids
It's just happening inside of Docker. Running on a local machine with exact the same files working like suposed to

### References
it looks similiar like this :

#15756
*

**Comment by jackofallops** [comment]
Hi @Aschenbecher 
Are you able to let us know which other providers and versions you mention are working?

**Comment by Aschenbecher** [comment]
Hello @jackofallops 
We are using the same setup with AWS and Oracle. And it was also working with azurem v1.x.
And i never saw terraform crashing ....

**Comment by jackofallops** [comment]
Hi @Aschenbecher 
I appreciate that; the providers are all quite different, and move at difference cadences with libraries, SDK's etc.  If you can let us know which versions of the providers you're using it this way I can look into the differences and try to track down if there's something we've changed or updated that the others haven't (or vice versa). 

As an aside - do you have the container RAM constrained? If so, does the same problem occur if you increase the available memory? (`signal SIGBUS: bus error code=0x2` suggests an issue with memory access)

**Comment by Aschenbecher** [comment]
Hello @jackofallops ,
Thanks for your fast reply!
Oracle provider Version: 3.72
AWS Provider: 2.61

I will double check the memory, but I'm sure I did it already and I dont changed the Host Server, But I will recheck it!

**Comment by Aschenbecher** [comment]
@jackofallops 
I just upgraded the Server instance from A2_v2 to A8_v2, still the same problem. Since A8_v2 has 16 GB of memory, this should not be the problem.....

**Comment by jackofallops** [comment]
> @jackofallops
> I just upgraded the Server instance from A2_v2 to A8_v2, still the same problem. Since A8_v2 has 16 GB of memory, this should not be the problem.....

I meant memory for the container itself, not the host - sorry for the confusion. I've done some testing of various versions of terraform and the AzureRM provider and not managed to recreate your issue.  I can't comment or assist with your Docker configuration, but that's where I'd start. Do you get the same outcome on Docker on your local machine?  (It may help to make use of https://hub.docker.com/r/hashicorp/terraform/)

**Comment by Aschenbecher** [comment]
@jackofallops 
Thanks for pointing me to this, I tested the same docker container locally and its working, maybe something with the our Azure network is broken since my upgrade of the terraform files to V2.x....

**Comment by jackofallops** [comment]
Happy to help - I'm going to close this issue for now, but if you do find something awry with the provider please do let us know.
Thanks!

**Comment by Aschenbecher** [comment]
I will also add the beginning of the logs now:

the local version:
```hcl
Acquiring state lock. This may take a few moments...
var.environment
  Enter a value: testing

var.num_servers
  Enter a value: 1

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.terraform_remote_state.network: Refreshing state...
data.azurerm_resource_group.metabuild: Refreshing state...
data.azurerm_image.optimizerimage: Refreshing state...
data.azurerm_image.simimage: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_linux_virtual_machine.optimizer will be created
  + resource "azurerm_linux_virtual_machine" "optimizer" {
      + admin_password                  = (sensitive value)
      + admin_username                  = "lisa"
      + allow_extension_operations      = true
```

The server version which crashes:
```hcl
var.environment
  Enter a value: testing

var.num_servers
  Enter a value: 1

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.terraform_remote_state.network: Refreshing state...
data.azurerm_resource_group.metabuild: Refreshing state...
data.azurerm_image.simimage: Refreshing state...
data.azurerm_image.optimizerimage: Refreshing state...

------------------------------------------------------------------------

Error: Unrecognized remote plugin message: 

This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol.


fatal error: fault
2020-06-17T14:37:43.580Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: [signal SIGBUS: bus error code=0x2 addr=0x360a7c0 pc=0x360a7c0]
2020-06-17T14:37:43.580Z [DEBUG] plugin.terraform-provider-azurerm_v2.14.0_x5: 
```
Do you know what is terraform trying to acces next after refreshing the statefiles?

**Comment by Aschenbecher** [comment]
Sorry, I reopened it accidently, because I was already writing my last post :(

**Comment by jackofallops** [comment]
> Sorry, I reopened it accidently, because I was already writing my last post :(

Not a problem.  Terraform lists the items in the state and instantiates the provider plugins to do refresh its view of the world.  I suspect your container is being bound on memory in some manner preventing the plugin doing so.

**Comment by Aschenbecher** [comment]
@jackofallops Hello Steve, after alot of debugging I found the problem which causes the problem. In the initial issue post I forgot one important information: the Terraform files are on a Azure SMB share. I didnt cared about it, because we used it already before without any problems. I can reproduce it when I copy the terraform files to my own NAS with smb, Terraform with the new Azure module will crash. I also tested it with the same setting with another provider (oracle) and I have no crashes. I was looking in terraform git issues for this problem and I found this: https://github.com/hashicorp/terraform/issues/17115
Hopefully this will help to fix this issue, because for our deployment it would be much better to use the SMB .
